model/Image.php

```diff
@@ -328,6 +342,97 @@ public function generateSetSize(Image_Backend $backend, $width, $height) {
 		return $backend->paddedResize($width, $height);
 	}
 
+	/**
+	 * Proportionally scale down this image if it is wider or taller than the specified dimensions.
+	 * Similar to SetRatioSize but without up-sampling. Use in templates with $ScaleMax.
+	 * @see Image::SetRatioSize()
+	 *
+	 * @param integer $width The maximum width of the output image
+	 * @param integer $height The maximum height of the output image
+	 * @return Image
+	 */
+	public function ScaleMax($width, $height) {
+		if ($this->width > $width || $this->height > $height) return $this->SetRatioSize($width,$height);
+		return $this;
+	}
+
+	/**
+	 * Proportionally scale down this image if it is wider than the specified width. 
+	 * Similar to SetWidth but without up-sampling. Use in templates with $ScaleMaxWidth.
+	 * @see Image::SetWidth()
+	 *
+	 * @param integer $width The maximum width of the output image
+	 * @return Image
+	 */
+	public function ScaleMaxWidth($width) {
+		if ($this->width > $width) return $this->SetWidth($width);
+		return $this;
+	}
+
+	/**
+	 * Proportionally scale down this image if it is taller than the specified height. 
+	 * Similar to SetHeight but without up-sampling. Use in templates with $ScaleMaxHeight.
+	 * @see Image::SetHeight()
+	 *
+	 * @param integer $height The maximum height of the output image
+	 * @return Image
+	 */
+	public function ScaleMaxHeight($height) {
+		if ($this->height > $height) return $this->SetHeight($height);
+		return $this;
+	}
+
+	/**
+	 * Crop this image to the aspect ratio defined by the specified width and height, 
+	 * then scale down the image to those dimensions if it exceeds them.
+	 * Similar to CroppedImage but without up-sampling. Use in templates with $CropMax.
+	 * @see Image::CroppedImage()
+	 *
+	 * @param integer $width The relative (used to determine aspect ratio) and maximum width of the output image
+	 * @param integer $height The relative (used to determine aspect ratio) and maximum height of the output image
+	 * @return Image
+	 */
+	public function CropMax($width, $height) {
+		// Is the image already the correct size?
+		if ($this->isSize($width, $height)) return $this;
+		
+		// If not, make sure the image isn't upsampled
+		$imageRatio = $this->width / $this->height;
+		$cropRatio = $width / $height;
+		// If cropping on the x axis compare heights
+		if ($cropRatio < $imageRatio && $this->height < $height) return $this->CroppedImage($this->height*$cropRatio, $this->height);
+		// Otherwise we're cropping on the y axis (or not cropping at all) so compare widths
+		if ($this->width < $width) return $this->CroppedImage($this->width, $this->width/$cropRatio);
```

Comment:
Ergh... Aargh... My brain is hurting too. Line 422 catches images that are going to be cropped on the x axis but are undersize and changes the resize arguments to suit. Line 424 catches images that are going to be cropped on the y axis AND images that aren't going to be cropped at all but are undersize. Checking the width only is sufficient at this point to cover both cases because we already know that `$cropRatio>= $imageRatio`. If the image isn't going to be cropped (the `$imageRatio==$cropRatio` scenario) it doesn't matter if we look at width or height, either one can tell us if the image is undersize so that is covered. So... I'm 96% confident the code doesn't need changing :) but, *sigh*, maybe we need more unit tests.